tests/dialects/test_spark.py

```diff
@@ -232,6 +232,20 @@ def test_spark(self):
                 "spark": "CAST(x AS BOOLEAN)",
             },
         )
+        self.validate_all(
+            "FLOAT(x)",
+            write={
+                "": "CAST(x AS FLOAT)",
+                "spark": "CAST(x AS FLOAT)",
+            },
+        )
```

Comment:
@GeorgeSittas  Should I add a dry up the tests too or is this fine for now?

Reply:
yea let's dry up the tests

Reply:
Cool! Will do in the morning

Reply:
Does this work? cfa50ae7fbbb3e21ee256de40eda45474f5862c8

https://github.com/tobymao/sqlglot/blob/cfa50ae7fbbb3e21ee256de40eda45474f5862c8/tests/dialects/test_spark.py#L229-L243

Opted for keeping the STRING case out of the loop since I felt it made the loop less clear, but could also do this or something 🤷🏻‍♂️ 

```python
        for data_type in ("BOOLEAN", "DATE", "DOUBLE", "FLOAT", "INT", "STRING", "TIMESTAMP"):
            self.validate_all(
                f"{data_type}(x)",
                write={
                    "": f"CAST(x AS {data_type if data_type != "STRING" else "TEXT"})",
                    "spark": f"CAST(x AS {data_type})",
                },
            )
```